service/storage_proxy.cc

```diff
@@ -3263,13 +3244,14 @@ storage_proxy::create_write_response_handler_helper(schema_ptr s, const dht::tok
     }
 
     // filter out natural_endpoints from pending_endpoints if the latter is not yet updated during node join
-    auto itend = boost::range::remove_if(pending_endpoints, [&natural_endpoints] (gms::inet_address& p) {
-        return boost::range::find(natural_endpoints, p) != natural_endpoints.end();
-    });
+    auto itend = std::ranges::remove_if(pending_endpoints, [&natural_endpoints] (gms::inet_address& p) {
+        return std::ranges::find(natural_endpoints, p) != natural_endpoints.end();
+    }).begin();
     pending_endpoints.erase(itend, pending_endpoints.end());
 
-    auto all = boost::range::join(natural_endpoints, pending_endpoints);
-    auto all_hids = all | boost::adaptors::transformed([&erm] (const gms::inet_address& ep) {
+    auto all = natural_endpoints;
+    std::ranges::copy(pending_endpoints, std::back_inserter(all));
```

Comment:
inet_address_vector_replica_set is small_vector of 3 element. If pending is not empty there will be a copy here.

Reply:
Yes, but it's rare.

Reply:
With tablets it will happen during tablet migration.

Reply:
Yes, but usually 2/100 of the tablets are migrating at any given time. So it's still very rare.

I could use a larger small_vector here if you think it's worthwhile.

Reply:
IDK. Larger small_vector will just waste memory in the common case. Lets take care of it if and when it shows in profile.

Reply:
This is local, so it doesn't waste memory. I don't think we'll see it in a profile though.

Reply:
This one is, but if you will change the type it will change it everywhere. And IIRC we hold one in the write handler as well.

Reply:
btw, in a multi-DC env, 3 is always not enough.